### DISCUSSION TITLE
time field "t"

### DISCUSSION THREAD

**Discussion by anath93** [ouster-lidar/ouster-ros#213] [Q&A] [answered]
I apologize if this answered before, but just trying to understand concept so I can use this in SLAM instead of going tf2 way, is this field ("t") offset from header.stamp.nanoseconds based on the current time stamp of lidar firing or how is this calculated ?

**Comment by Samahu** [comment, marked as answer]
Hi @anath93, `t` refers to the time of firing relative to the beginning of the frame. Refer to this discussion for more details: https://github.com/ouster-lidar/ouster-ros/discussions/206

**Reply by anath93** [reply]
@Samahu Thank you for replying, I did review that #206, can we say current_t = header.stamp.timestamp[0] - current_frame_firing_time in nanoseconds ?

**Reply by Samahu** [reply]
`point_fire_time_ts(absolute) = header.stamp.timestamp[0] + t`

**Reply by anath93** [reply]
Apologies if I am repeating,  "" point_fire_time_ts(absolute) -  header.stamp.timestamp[0] = t "" ? I am trying to find "t" equation, meaning how it is calculated in the SDK today and getting printed in PCL2 side.

**Reply by Samahu** [reply]
Please refer to the line: https://github.com/ouster-lidar/ouster-ros/blob/6a7693c551a34fc9f8dc47c8dda09cc64e2c57c0/src/os_ros.cpp#L213 which is pretty much the same as: `t = point_fire_time_ts(absolute) - header.stamp.timestamp[0]`

**Reply by anath93** [reply]
Thank you !
